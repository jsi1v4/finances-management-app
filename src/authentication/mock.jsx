import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        isAuth: true,
        userName: 'John Doe',
        authOn: jest.fn(),
        authOff: jest.fn()
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
