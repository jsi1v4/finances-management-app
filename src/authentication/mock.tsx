import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        isAuth: true,
        session: null,
        authOn: jest.fn(),
        authOff: jest.fn()
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
