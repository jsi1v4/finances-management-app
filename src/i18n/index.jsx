import React, { createContext, useState } from 'react';
import { useIntl } from 'react-intl';

import { INIT_LOCALE } from '../config';

export function useI18n() {
  const intl = useIntl();
  return (id, values) => {
    const message = intl.formatMessage({ id }, values);
    return <>{message}</>;
  };
}

export const I18nContext = createContext(null);

export function I18nProvider({ children }) {
  const [lang, setLanguage] = useState(INIT_LOCALE);
  const changeLang = (value) => {
    setLanguage(value);
  };

  return (
    <I18nContext.Provider
      value={{
        lang,
        changeLang
      }}
    >
      {children}
    </I18nContext.Provider>
  );
}

export default I18nProvider;
