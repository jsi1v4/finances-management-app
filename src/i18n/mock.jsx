import React from 'react';

import { I18nProvider } from './index';
import { I18nMessagesProvider } from './messages';

export function I18nMock({ children }) {
  return (
    <I18nProvider>
      <I18nMessagesProvider>{children}</I18nMessagesProvider>
    </I18nProvider>
  );
}

export default I18nMock;
