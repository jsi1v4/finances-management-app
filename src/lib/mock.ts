export class Auth {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  onAuthStateChanged(fn = () => {}) {
    return () => {};
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  async setPersistence(persist?: string) {
    return new Promise((resolve) => resolve({}));
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  async signInWithEmailAndPassword(username: string, password: string) {
    return new Promise((resolve) => resolve({}));
  }

  async signOut() {
    return new Promise((resolve) => resolve({}));
  }
}

export default Auth;
