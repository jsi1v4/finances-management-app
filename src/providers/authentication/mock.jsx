import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        userName: 'John Doe',
        signOut: jest.fn(),
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
