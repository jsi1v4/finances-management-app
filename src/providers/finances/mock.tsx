import React from 'react';

import { FinancesContext } from './index';
import { FinancesService } from './mock-service';

export function FinancesMock({ children }) {
  return (
    <FinancesContext.Provider value={new FinancesService({})}>
      {children}
    </FinancesContext.Provider>
  );
}

export default FinancesMock;
