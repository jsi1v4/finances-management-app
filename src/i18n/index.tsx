import React, { createContext, useContext, useState } from 'react';
import { useIntl } from 'react-intl';

import { INIT_LOCALE } from 'src/config';
import { II18nContext, Locale } from './types';

export const I18nContext = createContext<II18nContext>(null as never);

export const useI18n = () => useContext(I18nContext);

export function I18nProvider({ children }) {
  const [lang, setLanguage] = useState<Locale>(INIT_LOCALE);
  const changeLang = (value: Locale) => {
    setLanguage(value);
  };

  return (
    <I18nContext.Provider
      value={{
        lang,
        changeLang
      }}
    >
      {children}
    </I18nContext.Provider>
  );
}

export function useI18nMessage() {
  const intl = useIntl();
  return (id: string, values?: Record<string, string>) => {
    const message = intl.formatMessage({ id }, values);
    return message;
  };
}
