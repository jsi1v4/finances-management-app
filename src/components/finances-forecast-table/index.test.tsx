import React from 'react';
import { render } from '@testing-library/react';

import { I18nMock } from 'src/i18n/mock';
import { FinancesForecastTable } from './index';

const MOCK = [
  {
    key: '1',
    description: 'Test',
    yearOf: 2021,
    yearTo: 2021,
    monthOf: 1,
    monthTo: 2,
    cost: 1000
  }
];

describe('FinancesForecastTable component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<FinancesForecastTable data={[]} />, {
      wrapper: I18nMock
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component', async () => {
    const { getByText } = render(<FinancesForecastTable data={MOCK} />, {
      wrapper: I18nMock
    });

    expect(getByText(/Test/g).innerHTML).toBeDefined();
    expect(getByText(/Jan 2021/g).innerHTML).toBeDefined();
    expect(getByText(/Feb 2021/g).innerHTML).toBeDefined();
    expect(getByText(/1,000/g).innerHTML).toBeDefined();
  });
});
