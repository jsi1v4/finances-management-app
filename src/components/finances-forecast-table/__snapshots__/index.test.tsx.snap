// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancesForecastTable component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-card"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-small ant-table-bordered ant-table-empty"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 300px; min-width: 300px;"
                      />
                      <col
                        style="width: 120px; min-width: 120px;"
                      />
                      <col
                        style="width: 120px; min-width: 120px;"
                      />
                      <col
                        style="width: 120px; min-width: 120px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Description
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Date of
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Date to
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Cost
                        </th>
                        <th
                          class="ant-table-cell"
                        />
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="5"
                        >
                          <div
                            class="ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                class="ant-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    class="ant-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    class="ant-empty-img-simple-g"
                                    fill-rule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      class="ant-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <p
                              class="ant-empty-description"
                            >
                              No Data
                            </p>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="ant-card-actions"
    >
      <li
        style="width: 100%;"
      >
        <span>
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn ant-btn-link"
                type="button"
              >
                <span
                  aria-label="plus"
                  class="anticon anticon-plus"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="plus"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <defs>
                      <style />
                    </defs>
                    <path
                      d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                    />
                    <path
                      d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                    />
                  </svg>
                </span>
                <span>
                  Add
                </span>
              </button>
            </div>
          </div>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
